streams/streams-scala/src/main/scala/org/apache/kafka/streams/scala/StreamsBuilder.scala

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.scala
+
+import java.util.regex.Pattern
+
+import org.apache.kafka.streams.kstream.{GlobalKTable, Materialized}
+import org.apache.kafka.streams.processor.{ProcessorSupplier, StateStore}
+import org.apache.kafka.streams.state.StoreBuilder
+import org.apache.kafka.streams.{Consumed, StreamsBuilder => StreamsBuilderJ, Topology}
+
+import org.apache.kafka.streams.scala.kstream._
+import ImplicitConversions._
+import scala.collection.JavaConverters._
+
+/**
+  * Wraps the Java class StreamsBuilder and delegates method calls to the underlying Java object.
+  */
+class StreamsBuilder(inner: StreamsBuilderJ = new StreamsBuilderJ) {
```

Comment:
This and all other classes are public API. Thus, we should improve the JavaDocs for those classes and also add JavaDocs for all methods. I guess we can c&p from existing Java classes.

Reply:
@mjsax - Looking for suggestions. Should we copy/paste Javadoc from Java classes or use `@see` annotation ? The problem with copy is maintenance - when one changes someone needs to be careful enough to change the other.

Reply:
Good question. Not sure. I agree that maintaining JavaDocs twice is a hassle and error prone. But might be annoying for user if it's only linked on the other hand. Would be good to hear what others thing. \cc @guozhangwang @deanwampler @ijuma @vvcephei @bbejeck @seglo

Reply:
Adding JavaDocs to all the public API methods in the PR is the same amount of work any way we do it.  From an end user perspective I agree it would be nice to have the same (or slightly tweaked, as necessary) JavaDocs for all public API methods, plus a `@see` or `@link` tag to the corresponding Java API method.  It will be a small burden to maintain it going forward so I defer to the AK committers to make the call on the format.

Reply:
Some thoughts:
- I think the Consumer and KafkaConsumer pattern is bad. The documentation should have been on `Consumer` instead. The `AdminClient` follows the latter pattern.
- I think it's a poor user experience to ask users to read the docs in the Java class. My recommendation would be to at least include a short description in the Scala docs along with a link to the relevant Java documentation. The short description is less likely to change and it helps users make progress without having to jump to the Java code all the time. However, for more detailed information (which is more likely to change), they can check the Java code.

Reply:
I tend to agree with @ijuma comment about `KafkaConsumer`/`Consumer` pattern -- it's quite annoying to not get the JavaDocs directly. Thus, even if it's a burden it seems to be worth to maintain two copies.

Reply:
@mjsax - I have started writing the Scaladocs in the commit https://github.com/apache/kafka/pull/4756/commits/631ab9a8357e4667e4b941a0389bc4275f486711 .. Pls review if it's following the correct pattern